src/plugins/workspace/public/components/workspace_detail/association_data_source_modal.tsx

```diff
@@ -31,6 +31,9 @@ import { HttpStart, NotificationsStart, SavedObjectsStart } from '../../../../..
 import { AssociationDataSourceModalMode } from '../../../common/constants';
 import { Logos } from '../../../../../core/common';
 import { DirectQueryConnectionIcon } from '../workspace_form';
+import { DataConnectionIcon } from '../workspace_form';
+
+import { DATA_CONNECTION_SAVED_OBJECT_TYPE } from '../../../../data_source/common/data_connections';
 
 const ConnectionIcon = ({
```

Comment:
imo this could be cleaned up with using the data structures https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/common/datasets/types.ts#L75.

where we have the information about the icon and type

Reply:
Thanks, will take a look.

Reply:
I checked the interface of this data structure and haven't seen any related integration with data-connection type object. I suppose this is still in progress?